fili-core/src/test/groovy/com/yahoo/bard/webservice/data/dimension/DimensionRowSpec.groovy

```diff
@@ -26,4 +29,31 @@ class DimensionRowSpec extends Specification {
         testRow.getKeyValue() == keyValue
         testRow.getRowMap() == [(nonKeyFieldName): nonKeyValue, (keyFieldFieldName): keyValue]
     }
+
+    def "missing key throws error"() {
+        when:
+        DimensionRow testRow = new DimensionRow(missingKeyField, [(keyField): keyValue, (nonKeyField): nonKeyValue])
+
+        then:
+        thrown(IllegalArgumentException)
+    }
+
+    def "copyWithReplace copies and replaces"() {
```

Comment:
The name implies that it copies and replaces, but I don't see any validation that it actually copies (i.e. leaves the original unmodified). Should we include an additional assertion that `testRow != expectedRow`?

Reply:
done